src/workflow/restore.c

```diff
@@ -47,6 +49,10 @@ bool workflow_restore_backup(const RestoreBackupRequest* restore_request)
     if (!memory_set_initialized()) {
         return false;
     }
+    if (!workflow_confirm_time(restore_request->timestamp, restore_request->timezone_offset)) {
+        return false;
```

Comment:
this and the following (counter_set) should be before `memory_set_initialized`, otherwise you are in some semi/undefined state for U2F.

Reply:
ack